docs/man/man3/nxt_unit_request_read.3

```diff
@@ -0,0 +1,142 @@
+.\" (C) 2023, NGINX, Inc.
+.\"
+.TH nxt_unit_request_read 3 (date) "NGINX Unit (unreleased)"
+.SH Name
+nxt_unit_request_read
+\-
+read  content of a response in Unit app
+.SH Library
+NGINX Unit library
+.RI ( libunit ", " -lunit )
+.SH Synopsis
+.nf
+.B #include <nxt_unit.h>
+.PP
+.BI "ssize_t nxt_unit_request_read(nxt_unit_request_info_t *" req ,
+.BI "                              void " dst [. size "], size_t " size );
+.fi
+.SH Arguments
+.TP
+.I req
+Request object.
+.TP
+.I dst
+Buffer where the request message body will be written.
+It's a buffer of
+.I size
+bytes.
+.SH Description
+.MR nxt_unit_request_read 3
+copies the client request message body into a buffer.
+.PP
+This function does not read
+the header fields nor the request line,
+which are available in
+.IR req\->request .
```

Comment:
This function may also read less than the Content-Length if a data_handler callback is defined, in which case the full body may or may not be available in request_handler.

If it is available, then it can be read, and data_handler() will not be called.

If it is not available, then data_handler() will be called, at which point it will be available.

Reply:
Thanks!  Do you know if there's any way to know how much is available before actually attempting to read?  Or do you just read and then know if it's available?

Reply:
There isn't as far as I can see, although I'd like to know that myself as well.